pkg/controller/utils/infinispan/configuration/xsite.go

```diff
@@ -169,16 +170,23 @@ func getNodePortServiceHostPort(service *corev1.Service, k *util.Kubernetes, log
 	//oc create clusterrolebinding <name> -n ${NAMESPACE} --clusterrole=cluster-reader --serviceaccount=${NAMESPACE}:<account-name>
 	workerList := &corev1.NodeList{}
 
-	//select workers only. AFAIK, only the workers have the proxy to the pods
-	labelSelector := labels.SelectorFromSet(map[string]string{"node-role.kubernetes.io/worker": ""})
-
-	listOps := &client.ListOptions{
-		LabelSelector: labelSelector,
-	}
-	err := k.Client.List(context.TODO(), workerList, listOps)
+	//select workers first
+	req, err := labels.NewRequirement("node-role.kubernetes.io/worker", selection.Exists, nil)
 	if err != nil {
 		return "", 0, err
 	}
+	listOps := &client.ListOptions{
+		LabelSelector: labels.NewSelector().Add(*req),
+	}
+	err = k.Client.List(context.TODO(), workerList, listOps)
+
+	if err != nil || len(workerList.Items) == 0 {
+		// Fallback selecting everything
+		err := k.Client.List(context.TODO(), workerList, nil)
+		if err != nil || len(workerList.Items) == 0 {
+			return "", 0, err
```

Comment:
This will return empty err if no nodes is found. Some error handling in this path doesn't make sense